api/storage_service.cc

```diff
@@ -1346,7 +1368,7 @@ void set_snapshot(http_context& ctx, routes& r, sharded<db::snapshot_ctl>& snap_
             } else if (scrub_mode_str == "VALIDATE") {
                 scrub_mode = sstables::compaction_type_options::scrub::mode::validate;
             } else {
-                throw std::invalid_argument(fmt::format("Unknown argument for 'scrub_mode' parameter: {}", scrub_mode_str));
+                throw httpd::bad_param_exception(fmt::format("Unknown argument for 'scrub_mode' parameter: {}", scrub_mode_str));
```

Comment:
Have you check that nodetool still works as expected?
Don't forget the the main use of the API is the JMX/nodetool

Reply:
Yes, all scrub dtests (that use nodetool) passed.
I'll add this to this PR's description.

Reply:
just to be clear, you changed the error scenario, so you should make sure the error path still works.

Reply:
yes, nodetool fails as expected:
```
E               java.lang.IllegalStateException: Scylla API server HTTP GET to URL '/storage_service/keyspace_scrub/ks' failed: std::invalid_argument (Unknown argument for 'scrub_mode' parameter: FOO)
```